third_party/move/tools/move-linter/tests/model_ast_lints/equal_operands_in_pure_bin_op.move

```diff
@@ -0,0 +1,70 @@
+module 0xc0ffee::m {
+
+    struct Counter has key, store, drop { i: u64 }
+
+    struct NestedCounter has key, store, drop {
+        n: Counter
+    }
+
+    struct HyperNestedCounter has key, store, drop {
+        n: NestedCounter
+    }
+
+    public fun test1(x: u64) {
+        if (x % x == 2) {
```

Comment:
Add tests with constants (`2 == 2`, `TWO == 2` where `TWO` is a const 2).

Would be good to add support for `x[0] == x[0]` etc., but this requires supporting well known functions. Maybe add it to tests with a TODO saying these may be supported in the future.